Dockerfile

```diff
@@ -7,7 +17,7 @@ ARG NDCTL_BUILD_DEPS="os-core-dev devpkg-util-linux devpkg-kmod devpkg-json-c"
 
 #pull dependencies required for downloading and building libndctl
 ARG CACHEBUST
-RUN swupd update && swupd bundle-add ${NDCTL_BUILD_DEPS} go-basic-dev && rm -rf /var/lib/swupd
+RUN swupd update ${SWUPD_UPDATE_ARG} && swupd bundle-add ${NDCTL_BUILD_DEPS} go-basic-dev && rm -rf /var/lib/swupd
```

Comment:
@pohly if you have already chosen a specific CLEAR_LINUX_BASE why do we the the update. If you are choosing the version you are going to update to, then what is the need for CLEAR_LINUX_BASE.

Is the fact that you know you are running `nnn` post update not enough.

Is this just so that you know exactly what happened? i.e. you picked the latest clearlinux container image which was hash `xyz` and you updated it to `nnn`

Reply:
> if you have already chosen a specific CLEAR_LINUX_BASE why do we the the update.

Because the Clear Linux base image only gets updated once a month. Releasing without "swupd update" would lead to an app with a potentially insecure OS and isn't acceptable.

>  If you are choosing the version you are going to update to, then what is the need for CLEAR_LINUX_BASE.

It's necessary to ensure reproducibility. We cannot start with "clearlinux:latest" and the do "swupd update -m <someversion>" because that breaks once the base image is on a newer version than `<someversion>`, as "swupd updated" doesn't support downgrades. The Clear Linux stacks team is doing that at the moment, and older tags in their repos cannot be built anymore because of that (I checked).

Reply:
I need to point out one assumption: this approach relies on those `clearlinux@sha256:....` images remaining available even when they are no longer referenced by a named tag. I can't test that at the moment because I don't have a hash for an older image.

If they don't remain available, then this approach is no worse than referencing `clearlinux:latest` because it works as long as that tag is set.